process/block/preprocess/transactions.go

```diff
@@ -471,9 +473,175 @@ func (txs *transactions) CreateAndProcessMiniBlocks(
 		}
 	}
 
+	compressedMiniBlocks := txs.compressMiniBlocks(miniBlocks)
+	if len(compressedMiniBlocks) > 0 {
+		return compressedMiniBlocks, nil
+	}
+
 	return miniBlocks, nil
 }
 
+func (txs *transactions) compressMiniBlocks(miniBlocks block.MiniBlockSlice) block.MiniBlockSlice {
```

Comment:
Extract all these methods in a new struct! 
You added around 200 lines of code here.

Reply:
+1

Reply:
Refactored